src/Examples/Program.cs

```diff
@@ -77,33 +84,96 @@ internal static async Task MainAsync(string[] args)
+                    // create game
+                    var gameProcess = await _gameFactory.CreateSingleAsync(new ZSingleParams { Game = game });
 
-                    // run game process
-                    var runResult = await gameProcess.RunAsync();
-
-                    Console.WriteLine($"Run result {runResult}");
+                    // run and track game pipe
+                    await _RunAndTrack(gameProcess);
 
                     break;
                 case ZPlayMode.Multiplayer:
+
+                    await _MultiplayerHandler(game);
+
                     break;
 
                 case ZPlayMode.CooperativeHost:
                 case ZPlayMode.CooperativeClient:
                 case ZPlayMode.TestRange:
+                    //testrange process 
+             var testRange = await _gameFactory.CreateTestRangeAsync(new ZTestRangeParams { Game = game });
+
+                  //run test range process handler 
+             await _RunAndTrack(testRange);
+
+                    break;
                 default:
                     throw new ArgumentOutOfRangeException();
             }
         }
+
+        internal static async Task _RunAndTrack(IZRunGame gameProcess)
+        {
+            var resetEvent = new ManualResetEvent(false);
+
+            // track game pipe
+            gameProcess.Pipe += (sender, pipeArgs) =>
+            {
+                Console.WriteLine(pipeArgs.FullMessage);
+
+                // return from _RunAndTrack if game closed
+                if (pipeArgs.SecondPart.Contains("Closed"))
+                {
+                    resetEvent.Set();
+                }
+            };
+
+            // run game process
+            var runResult = await gameProcess.RunAsync();
+
+            // the result will be an enumeration
+            // that will help determine if the game was launched successfully (returned directly by the ZClient)
+            if (runResult != ZRunResult.Success)
+            {
+                // TODO: Do some stuff here
+            }
+
+            resetEvent.WaitOne();
+        }
+
+        internal static async Task _MultiplayerHandler(ZGame game)
+        {
+            //table line for top and bottom 
+            string straithLine = "_______________________________________________________________________________________";
+            var service = _zloApi.CreateServersListService(game);
+            var factory = _zloApi.GameFactory;
+            ManualResetEvent resetEvent = new ManualResetEvent(false);
+            service.InitialSizeReached += (s, e) => resetEvent.Set();
+
+            service.StartReceiving();
+            resetEvent.WaitOne();
+
+            Console.ForegroundColor = ConsoleColor.DarkYellow;
+            Console.WriteLine("\n {0,88} \n {1,5} {2,50}| {3,20}| \n {4,88}", straithLine, "ID:", "SERVERNAME", "MAP:", straithLine);
+            foreach (var item in service.ServersCollection)
```

Comment:
@nintend01337 Please separate logically different code blocks with line breaks
For example:
```
Console.WriteLine(...);
Console.Write(...);
Console.Foreground = ...;
// use break line here
foreach (var x in collection ) { ... }